### PR TITLE
created data access interface, checkstyled edit use case

### PR DESCRIPTION
**Pull request by venniseho**
as the title says, I created a data access interface that aligns with our change to being dependent on data access interface instead of schedule itself (dependency inversion principle) and I fixed checkstyle for all my edit files in use case.

### CODE REVIEW

File: src/main/java/usecase/edit/EditDataAccessInterface.java
Comment (utcjcao) [inline src/main/java/usecase/edit/EditDataAccessInterface.java:1]: might want to clarify that edit data access interface is an interface in javadoc, also may want to clarify where it is used

**Review by utcjcao — COMMENTED** [review]
Looks good!

May want to clarify tests for edit interactor. should use the full model (input boundary, output boundary, etc)

**Review by RoyceLuo1216 — CHANGES_REQUESTED** [review]
Make sure to correct changes for data memory access!

**Review by RoyceLuo1216 — APPROVED** [review]
Good changes

**Pull request merged**